docker/android/android/soong.py

```diff
@@ -0,0 +1,654 @@
+'''
```

Comment:
This is a complete and **mostly** correct Soong parser. It will accept all valid input, and reject most invalid input. It however, converts it to an AST, but does not do validation of that AST, only the initial grammar, so identifiers of incorrect types may not be properly processed. This is irrelevant here, since we only want to remove tests from the file and then re-serialize it: not validate and re-implement the entire build system.